packages/phoenix-event-display/src/tests/managers/ui-manager/dat-gui-ui.test.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * @jest-environment jsdom
+ */
+import { InfoLogger } from '../../../helpers/info-logger';
+import { ThreeManager } from '../../../managers/three-manager';
+import { DatGUIMenuUI } from '../../../managers/ui-manager/dat-gui-ui';
+import { GUI } from 'dat.gui';
+import { Object3D } from 'three';
+
+jest.mock('../../../managers/three-manager');
+
+describe('DatGUIMenuUI', () => {
+  let datGUIMenu: DatGUIMenuUI;
+  let datGUIMenuPrivate: any;
+
+  beforeEach(() => {
+    datGUIMenu = new DatGUIMenuUI(
+      'eventDisplay',
+      new ThreeManager(new InfoLogger())
+    );
+    datGUIMenuPrivate = datGUIMenu as any;
+  });
+
+  afterEach(() => {
+    datGUIMenu.clear();
+  });
+
+  it('should create an instance', () => {
+    expect(datGUIMenu).toBeTruthy();
+  });
+
+  it('should add geometry (detector geometry) folder to the menu', () => {
+    datGUIMenu.addGeometryFolder();
+
+    expect(datGUIMenuPrivate.geomFolder).toBeInstanceOf(GUI);
+    expect(datGUIMenuPrivate.geomFolder.name).toBe('Geometries');
+  });
+
+  it('should add geometry to the menu geometry folder and set up its configurable options', () => {
+    datGUIMenu.addGeometryFolder();
+    datGUIMenu.addGeometry(new Object3D(), 'test');
+
+    expect(datGUIMenuPrivate.geomFolder.__folders).toBeInstanceOf(Object);
+  });
+
+  it('should add event data folder with functions for event data toggles like show/hide and depthTest', () => {
+    datGUIMenu.addEventDataFolder();
+
+    expect(datGUIMenuPrivate.eventFolder).toBeInstanceOf(GUI);
+    expect(datGUIMenuPrivate.eventFolder.name).toBe('Event Data');
+    expect(datGUIMenuPrivate.eventFolder.__folders).toEqual({});
+  });
+
+  it('should add folder for event data type like tracks or hits to the menu', () => {
+    datGUIMenu.addEventDataFolder();
+    datGUIMenu.addEventDataTypeFolder('Event Data');
+
+    expect(
+      datGUIMenuPrivate.eventFolder.__folders['Event Data']
+    ).toBeInstanceOf(GUI);
+  });
+
+  it('should add collection folder and its configurable options to the event data type (tracks, hits etc.) folder', () => {
+    datGUIMenu.addEventDataFolder();
+    datGUIMenu.addCollection('Event Data', 'test');
+
+    expect(
+      datGUIMenuPrivate.eventFolder.__folders['Event Data']
```

Comment:
It seems this doesn't check if the collection folder was added or not.

Reply:
It will be added but yeah this test case is not explicit enough. Removing it.